editor/src/clj/editor/app_view.clj

```diff
@@ -1735,7 +1735,10 @@ If you do not specifically require different script states, consider changing th
            (let [^SplitPane editor-tabs-split (g/node-value app-view :editor-tabs-split)
                  tab-panes (.getItems editor-tabs-split)
                  open-tabs (mapcat #(.getTabs ^TabPane %) tab-panes)
-                 view-type (if (g/node-value resource-node :editable) view-type text-view-type)
+                 view-type (if (and (= :code (:id view-type))
+                                    (not (g/node-value resource-node :editable)))
+                             text-view-type
+                             view-type)
```

Comment:
Previously the only "non-editable" resources used the code view. Now we also use it for non-editable game objects and collections, which obviously shouldn't be opened in text mode.